sdk/src/main/scala/com/horizen/chain/package.scala

```diff
@@ -4,11 +4,11 @@ import com.horizen.utils.ByteArrayWrapper
 import supertagged.TaggedType
 
 package object chain {
-  object MainchainBlockReferenceId extends TaggedType[ByteArrayWrapper]
-  type MainchainBlockReferenceId = MainchainBlockReferenceId.Type
-  val mainchainBlockReferenceIdSize = 32
+  object MainchainBlockReferenceHash extends TaggedType[ByteArrayWrapper]
```

Comment:
The concept of `MainchainBlockReferenceId` is not relevant anymore.
`MainchainBlockReference` has `id()`, `header.hash` and `dataHash`.
`MainchainBlockReferenceId` is actually a `header.hash`.
This change is just a minimum required one, but not enough. The usage of `MainchainBlockReferenceId/MainchainBlockReferenceHash` must be reviewed deeper,

Reply:
done